### PR TITLE
Fix/safety cap

### PR DESCRIPTION
**Pull request by JhChoy**
- Add byte safety cap
- Currently in zkSync, it appears that the leftover bytes do not disappear even if you cast from uint256 to uint64 via assembly and then return from the function.

### CODE REVIEW

File: src/libraries/FeePolicy.sol [hunk 1]
```diff
@@ -22,7 +22,7 @@ library FeePolicyLibrary {
 
         uint256 mask = usesQuote_ ? 1 << 23 : 0;
         assembly {
-            feePolicy := or(mask, add(rate_, MAX_FEE_RATE))
+            feePolicy := or(mask, add(and(rate_, 0xffffff), MAX_FEE_RATE))
```
Comment (detectivekim) [inline src/libraries/FeePolicy.sol:25]: duplicate AND

File: src/libraries/SignificantBit.sol [hunk 2]
```diff
@@ -19,12 +19,11 @@ library SignificantBit {
         assembly {
             index := shr(248, mul(and(x, add(not(x), 1)), DEBRUIJN_SEQ))
         }
-        return uint8(DEBRUIJN_INDEX[index]); // can optimize with CODECOPY opcode
+        return uint8(DEBRUIJN_INDEX[index] & 0xff); // can optimize with CODECOPY opcode
```
Comment (detectivekim) [inline src/libraries/SignificantBit.sol:22]: Is it really necessary?

File: src/libraries/SignificantBit.sol [hunk 3]
```diff
@@ -49,7 +48,8 @@ library SignificantBit {
             x := shr(f, x)
             f := gt(x, 0x1)
             msb := or(msb, f)
+
+            msb := and(msb, 0xff)
```
Comment (detectivekim) [inline src/libraries/SignificantBit.sol:52]: Is it really necessary?

File: src/libraries/FeePolicy.sol (hunk 1 above)
Comment (JhChoy) [reply src/libraries/FeePolicy.sol:25]: Why is it duplicated?

**Review by detectivekim — APPROVED** [review]

**Pull request merged**